test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/scaling/SmartRouterLoadBalancing.java

```diff
@@ -0,0 +1,92 @@
+
+    @Override protected WorkbenchRuntimeSmartRouterKieServerDatabaseScenario createDeploymentScenario(DeploymentScenarioBuilderFactory deploymentScenarioFactory) {
+        return deploymentScenarioFactory
+                .getWorkbenchRuntimeSmartRouterKieServerDatabaseScenarioBuilder()
+                .withExternalMavenRepo(MavenConstants.getMavenRepoUrl(), MavenConstants.getMavenRepoUser(), MavenConstants.getMavenRepoPassword())
+                .build();
+    }
+
+    @Test
+    public void testRouterLoadBalancing() {
+        MavenDeployer.buildAndDeployMavenProject(ClassLoader.class.getResource("/kjars-sources/definition-project-snapshot").getFile());
+
+        kieControllerClient = KieServerControllerClientProvider.getKieServerMgmtControllerClient(deploymentScenario.getWorkbenchRuntimeDeployment());
+        kieServerClient = KieServerClientProvider.getKieServerClient(deploymentScenario.getKieServerDeployment());
+
+        KieServerInfo serverInfo = kieServerClient.getServerInfo().getResult();
+        kieControllerClient.saveContainerSpec(serverInfo.getServerId(), serverInfo.getName(), CONTAINER_ID, CONTAINER_ALIAS, PROJECT_GROUP_ID, DEFINITION_PROJECT_SNAPSHOT_NAME, DEFINITION_PROJECT_SNAPSHOT_VERSION, KieContainerStatus.STARTED);
+        KieServerClientProvider.waitForContainerStart(deploymentScenario.getKieServerDeployment(), CONTAINER_ID);
+
+        deploymentScenario.getKieServerDeployment().scale(2);
+
+        kieServerClientRouter = KieServerClientProvider.getSmartRouterClient(deploymentScenario.getSmartRouterDeployment(),
+                deploymentScenario.getKieServerDeployment().getUsername(), deploymentScenario.getKieServerDeployment().getPassword());
+
+        ServiceResponse<KieServerInfo> kieServerInfo = kieServerClientRouter.getServerInfo();
+        List<String> capabilities = kieServerInfo.getResult().getCapabilities();
+        Assertions.assertThat(capabilities).isNotEmpty();
+
+        QueryServicesClient queryServicesClient = kieServerClientRouter.getServicesClient(QueryServicesClient.class);
+        List<ProcessDefinition> processDefinitions = queryServicesClient.findProcesses(0, 100);
+        Assertions.assertThat(processDefinitions).isNotNull();
+        Assertions.assertThat(processDefinitions.stream().anyMatch(p -> p.getId().equals(LOG_PROCESS_ID)));
+
+        ProcessServicesClient processServicesClient = kieServerClientRouter.getServicesClient(ProcessServicesClient.class);
+        for (int i = 0; i < PROCESS_NUMBER; i++) {
+            processServicesClient.startProcess(CONTAINER_ID, LOG_PROCESS_ID);
+        }
+
+        List<Instance> kieServerInstances = deploymentScenario.getKieServerDeployment().getInstances();
+        Assertions.assertThat(kieServerInstances.size()).isGreaterThan(1);
+        for (Instance kieServerInstance : kieServerInstances) {
+            Assertions.assertThat(kieServerInstance.getLogs()).contains(LOG_MESSAGE);
```

Comment:
As far as load balancing goes, wouldn't it be better to check that the instances have been distributed more or less evenly (say by expecting that each instance has "at least 30%" of messages)?